### ISSUE TITLE
8.0.1 hosting bundle no longer allows net8.0-windows

### ISSUE THREAD

**Issue by SimonCropp** [dotnet/sdk#37892]
Our projects target `net8.0-windows`

```
<TargetFramework>net8.0-windows</TargetFramework>
```

After updating to sdk 8.0.101 and rebuilding our apps no longer run:

```
You must install or update .NET to run this application. 
App: C:\Octopus\Applications\Dev\Notifications.Deploy\2.0.63603_2\installer\Notifications.Install.exe 
Architecture: x64 
Framework: 'Microsoft.WindowsDesktop.App', version '8.0.0' (x64) 
.NET location: C:\Program Files\dotnet\ 
No frameworks were found. 
Learn more: 
https://aka.ms/dotnet/app-launch-failed 
To install missing framework, download: 
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.WindowsDesktop.App&framework_version=8.0.0&arch=x64&rid=win-x64&os=win10 
```

note that 
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.WindowsDesktop.App&framework_version=8.0.0&arch=x64&rid=win-x64&os=win10 
redirects to 
https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-desktop-8.0.1-windows-x64-installer?cid=getdotnetcore

When building from 8.0.100 sdk we used to be able to run with only the hosting bundle installed 
https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-aspnetcore-8.0.1-windows-hosting-bundle-installer
ie no Desktop Runtime required

if i change to changing `net8`

```
<TargetFramework>net8.0</TargetFramework>
```

and our apps start running again

## dotnet -- info

```
C:\Users\scropp>dotnet --info

Host:
  Version:      8.0.1
  Architecture: x64
  Commit:       bf5e279d92
  RID:          win-x64

.NET SDKs installed:
  No SDKs were found.

.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found
```

## runtimeconfig.json

and when i look the same runtimeconfig.json between the sdks the 8.0.1 has indeed added `Microsoft.WindowsDesktop.App`

#### from 8.0.100 sdk
```
{
  "runtimeOptions": {
    "tfm": "net8.0",
    "rollForward": "LatestMinor",
    "frameworks": [
      {
        "name": "Microsoft.NETCore.App",
        "version": "8.0.0"
      },
      {
        "name": "Microsoft.AspNetCore.App",
        "version": "8.0.0"
      }
    ]
  }
}
```

#### from 8.0.101 sdk

```
{
  "runtimeOptions": {
    "tfm": "net8.0",
    "rollForward": "LatestMinor",
    "frameworks": [
      {
        "name": "Microsoft.NETCore.App",
        "version": "8.0.0"
      },
      {
        "name": "Microsoft.WindowsDesktop.App",
        "version": "8.0.0"
      },
      {
        "name": "Microsoft.AspNetCore.App",
        "version": "8.0.0"
      }
    ]
  }
}
```

## Questions

 * So was this an intentional change?
 * Should we also be deploying Microsoft.WindowsDesktop.App to run our apps?

**Comment by baronfel** [comment]
@SimonCropp what's a good repro for this behavior? Is a new console app targeting `net8.0-windows` on 8.0.101 SDK all that's required to reproduce? I'm trying to get a binlog so I can look at how/why the set of `RuntimeFrameworks` passed to the `GenearteRuntimeConfigurationFiles` Task changes across versions.

**Comment by campersau** [comment]
I think I found a minimal reproduction, it looks like it is because of `Microsoft.Data.SqlClient`.
Create a new console app with this csproj:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0-windows</TargetFramework>
    <RuntimeIdentifier>win-x64</RuntimeIdentifier>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.Data.SqlClient" Version="5.1.4" />
  </ItemGroup>

</Project>
```

This will include the `Microsoft.WindowsDesktop.App` reference:
```json
{
  "runtimeOptions": {
    "tfm": "net8.0",
    "frameworks": [
      {
        "name": "Microsoft.NETCore.App",
        "version": "8.0.0"
      },
      {
        "name": "Microsoft.WindowsDesktop.App",
        "version": "8.0.0"
      }
    ],
    "configProperties": {
      "System.Reflection.Metadata.MetadataUpdater.IsSupported": false,
      "System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization": false
    }
  }
}
```

Do a `dotnet publish` then run it on a machine which only has the runtime or aspnet core hosting bundle installed.

**Comment by campersau** [comment]
Updating `Microsoft.Data.SqlClient` from `5.1.3` to `5.1.4` adds these dependencies and it looks like these are the actual issue here   at least for me (e.g. WebView2):

![image](https://github.com/dotnet/sdk/assets/4009570/49053b28-ff7e-4ae3-970a-cf7d31d569eb)

After I downgrade to `5.1.3` it works for me like before.

**Comment by baronfel** [comment]
@campersau I'm having a hard time reproducing. For this project:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0-windows</TargetFramework>
    <RootNamespace>sdk_37892</RootNamespace>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <RuntimeIdentifier>win-x64</RuntimeIdentifier>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.Data.SqlClient" Version="5.1.4" />
  </ItemGroup>
</Project>
```

when I publish I get the following runtimeconfig.json:

```json
{
  "runtimeOptions": {
    "tfm": "net8.0",
    "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "8.0.0"
    },
    "configProperties": {
      "System.Reflection.Metadata.MetadataUpdater.IsSupported": false,
      "System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization": false
    }
  }
}
```

without the WindowsDesktop framework.  I'll try again with the 5.1.3 to 5.1.4 transition and see if I can get it to trigger.

**Comment by baronfel** [comment]
We need to reach out to the IdentityClient folks to see if this is an absolute requirement. Their repository is 
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet.

**Comment by campersau** [comment]
Looks like there is already an issue for it: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4468

**Comment by baronfel** [comment]
In the interim, I think I'm curious why the behavior seemingly changed between 8.0.100 and 8.0.101 - @SimonCropp are you sure that there were no other package version changes as part of your update?

**Comment by SimonCropp** [comment]
@baronfel sorry you are correct. we also update the sql nuget when doing the sdk change. so i have my wires crossed

do u want to close this one? or rename it?

**Comment by baronfel** [comment]
I think we should close it for now - though we on the SDK team will try to reach out to the identity folks to determine a way to use the library that doesn't silently introduce new runtimes.

**Comment by SimonCropp** [comment]
ok will close. 

but is this effectively the cause of the bug https://github.com/dotnet/SqlClient/issues/1108#issuecomment-1828924151  ?

**Comment by baronfel** [comment]
That seems correct, looks like the team has a POC for removing some of the dependencies that may fix the issue?

**Comment by SimonCropp** [comment]
@baronfel is there a url for that discussion?

**Comment by rclabo** [comment]
This bit me and it's a pain.  I can't believe our website deployment that uses SqlServer is down because `Microsoft.Data.SqlClient` now requires a desktop framework install.  What?    

This title seems accurate and unthinkable at the same time. "8.0.1 hosting bundle no longer allows net8.0-windows"  It's hard to understand how such a bug could be released.  That it was released clearly shows the limits of unit testing.  And the bug still is not fixed after being identified 3.5 months ago.  (sigh)

Is this the best place to see the current status? https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4468  Are their other places one can get a status update?

**Comment by SimonCropp** [comment]
@rclabo here is the underlying issue and best place to discuss https://github.com/dotnet/SqlClient/issues/1108
